dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py

```diff
@@ -1226,6 +1230,44 @@ def assign_variables_indexes(self):
         self.nb_integer_vars = index_integer_double
         self.find_calculated_variables()
 
+    def detect_z_only_used_internally(self):
+        map_num_eq_vars_defined = self.get_map_num_eq_vars_defined()
+        dicr_variable_to_dics_equation_dependencies = {}
+
+        discr_vars = []
+        for var in self.list_vars:
+            name = var.get_name()
+            if name not in map_var_name_2_addresses : continue
+            test_param_address(name)
+            address = map_var_name_2_addresses[name]
+            if "discreteVars" in address or "integerDoubleVars" in address or "booleanVars" in address:
+                discr_vars.append(name)
```

Comment:
As it is used only once today I propose to add a function to compute it. If it becomes widely used then we'll add an attribute to the class. Is it OK?

Reply:
The function is fine for me. 

My point was a bit larger than just the implementation. I was wondering if there aren't existing structures in the Python scripts that would enable to build the dico without having to redo the testing here, like if we already have a list of discreteVars, integerDoubleVars and booleanVars with their names somewhere.
I suppose not, otherwise you could improve this later on.